clean-data/clean-data.R

```diff
@@ -0,0 +1,700 @@
+        mutate(ExpectedEarning = ExpectedEarning * 1000) %>%
+        mutate(ExpectedEarning = as.character(ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% setdiff(below100) %>% bind_rows(change100)
+
+    cleanPart1
+}
+
+# Normalize text based on searching list and desired single replacement
+# Using non-standard eval in dplyr: http://stackoverflow.com/a/26003971
+# More on NSE:
+#   https://cran.r-project.org/web/packages/dplyr/vignettes/nse.html
+normalize_text <- function(inData, columnName, searchTerms, replaceWith) {
+    # Setup dynamic naming of variables later in function
+    varval <- lazyeval::interp(~ replaceText, replaceText = replaceWith)
+
+    # Gets indices for rows that need to be changed
+    wordIdx <- inData %>% select_(columnName) %>%
+        mutate_each(
+            funs(grepl(paste(searchTerms, collapse = "|"),
+                       .,
+                       ignore.case = TRUE)
+            )
+        ) %>% unlist(use.names = FALSE)
+
+    # Change row values to intended words
+    wordData <- inData %>% filter(wordIdx) %>%
+        mutate_(.dots = setNames(list(varval), columnName))
+
+    # Combine data back together
+    cleanData <- inData %>% filter(!wordIdx) %>% bind_rows(wordData)
+
+    cleanData
+}
+
+# Take a range string (e.g. "50-60") and take average (e.g. "55").
+average_string_range <- function(x) {
+    avgRange <- c()
+    for (i in 1:length(x)) {
+        tempRange <- x[i] %>% strsplit("-|to") %>%
+            unlist %>%
+            as.numeric %>%
+            mean %>%
+            as.character()
+        avgRange <- c(avgRange, tempRange)
+    }
+    avgRange
+}
+
+# Remove non-numeric characters and change years to months
+years_to_months <- function(x) {
+    monthsDat <- c()
+    for (i in 1:length(x)) {
+        tempMonths <- x[i] %>% gsub("[A-Za-z ]", "", .) %>%
+            (function (x) as.numeric(x) * 12) %>%
+            as.character()
+        monthsDat <- c(monthsDat, tempMonths)
+    }
+    monthsDat
+}
+
+# Remove outliers based on threshold
+remove_outlier <- function(x, thres) {
+    ifelse(test = x >= thres, yes = as.numeric(NA), no = x)
+}
+
+# Clean Part 1 of survey
+clean_part_1 <- function(part1) {
+    # Job Role Interests
+
+    ## Title case answers for other job interests
+    ##  See if I can simplify this by just mutating
+    jobRoleOtherYes <- part1 %>% filter(!is.na(JobRoleInterestOther)) %>%
+        mutate(JobRoleInterestOther = simple_title_case(JobRoleInterestOther))
+    jobRoleOtherNo <- part1 %>% filter(is.na(JobRoleInterestOther))
+    cleanPart1 <- jobRoleOtherNo %>% bind_rows(jobRoleOtherYes)
+
+    ## Change uncertain job roles to "Undecided"
+    ##  Adapted from: http://stackoverflow.com/a/26766945
+    undecidedWords <- c("not sure", "don't know", "not certain",
+                        "unsure", "dont know", "undecided",
+                        "no preference", "not", "any", "no idea")
```

Comment:
You missed "all of the above" :)

Btw, how about adding Game Developer, Cyber Security and Software Engineer? These are the most common answers for "other".

gameDevWords <- c("game", "games")
cyberSecurityWords <- c("secure","security","penetration tester","PenTester")
softwareEngineerWords <- c("software")